pkg/peer/types.go

```diff
@@ -47,15 +60,21 @@ func (s SimulcastLayer) String() string {
 	}
 }
 
+// Basic information about a track.
 type TrackInfo struct {
 	TrackID  string
 	StreamID string
 	Codec    webrtc.RTPCodecCapability
 }
 
-type SimulcastTrackInfo struct {
+// Information with some extended (optional fields).
+// Ideally we would want to have different types for different tracks, but Go does not have ADTs,
+// so it's the only way to do it without writing too much of a boilerplate.
+type ExtendedTrackInfo struct {
+	// Information about a track.
 	TrackInfo
-	Layer SimulcastLayer
+	// Optional simulcast layer (if any). Would be nil for non-simulcast or audio tracks.
+	Layer *SimulcastLayer
```

Comment:
I don't like the usage of a pointer here... Could we add `SimulcastLayer.Invalid` or something along those lines?

Reply:
Yes, that's what I did here: https://github.com/matrix-org/waterfall/pull/85/commits/59579be99a6879efbb6a342d2ed864d5147f596b

Although it's a kludge for missing ADTs, it seems like in Go they oftentimes use pointers to signify such things.